### PR TITLE
Remove all calls to goog.dom.createDom

### PR DESCRIPTION
**Pull request by NeilFraser**
All XML nodes are now created using `Blockly.Xml.utils.createElement`.  All browser DOM nodes are now created using `document.createElement`.

### CODE REVIEW

**Comment by NeilFraser** [comment]
FYI, when the dependencies change, Travis' error message is completely useless.  It's a random code dump from Travis' codebase.  Recompiling was a wild guess, and it worked.

File: core/mutator.js [hunk 1]
```diff
@@ -33,8 +33,8 @@ goog.require('Blockly.Events.Ui');
 goog.require('Blockly.Icon');
 goog.require('Blockly.utils');
 goog.require('Blockly.WorkspaceSvg');
-
-goog.require('goog.dom');
+goog.require('Blockly.Xml');
```
Comment (rachel-fenichel) [inline core/mutator.js]: Why do you need both of these requires?

File: core/xml_utils.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+goog.provide('Blockly.Xml.utils');
+
+
+/**
+ * Create DOM element for XML.
+ * @param {string} tagName Name of DOM element.
+ * @return {!Element} New DOM element.
+ */
+Blockly.Xml.utils.createElement = function(tagName) {
+  // TODO: Namespace this element.
```
Comment (rachel-fenichel) [inline core/xml_utils.js]: Todos should refer to github issue numbers.

File: core/xml_utils.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+ * @name Blockly.Xml.utils
+ * @namespace
+ */
+goog.provide('Blockly.Xml.utils');
+
+
+/**
+ * Create DOM element for XML.
+ * @param {string} tagName Name of DOM element.
+ * @return {!Element} New DOM element.
```
Comment (rachel-fenichel) [inline core/xml_utils.js]: Mark all functions @public if you want to expose them fully, otherwise @package.

File: core/mutator.js (hunk 1 above)
Comment (NeilFraser) [reply core/mutator.js]: This file contains two calls to `Blockly.Xml.utils.createElement` and two calls to `Blockly.Xml.domToText`.  I could just require `Blockly.Xml` and rely on the (current) transitive dependency, but that's fragile.

Comment (rachel-fenichel) [reply core/mutator.js]: Okay.  Leaving it as-is sounds good to me.

**Pull request merged**